backend_jobs/ingestion_pipeline/providers/image_provider_flickr.py

```diff
@@ -36,65 +38,78 @@ def __init__(self, query_arguments=None):
-            #Extracts width from element, if the width equals str converts to int.
-            width_pixels = int(element.get('width_c')) if element.get('width_c') else None,
-            #Extracts height from element, if the height equals str converts to int.
-            height_pixels = int(element.get('height_c')) if element.get('height_c') else None)
+        image_attributes = ImageAttributes(
+            url=element.get('url_c'),
+            image_id=None,
+            image_type=ImageType.CAMERA,
+            date_shot=_get_date(element),
+            format=element.get('originalformat'),
+            attribution=element.get('ownername'),
+            latitude=float(element.get('latitude')),
+            longitude=float(element.get('longitude')),
+            # Extracts width from element, if the width equals str converts to int.
+            width_pixels=int(element.get('width_c')) if element.get('width_c') else None,
+            # Extracts height from element, if the height equals str converts to int.
+            height_pixels=int(element.get('height_c')) if element.get('height_c') else None)
         return image_attributes
 
     def get_url_for_min_resolution(self,  min_height, min_width, image):
         # This function modifies the url to get the requested resolution,
         # in flickr the resolution is represented in the url.
         # See details at https://www.flickr.com/services/api/misc.urls.html
         url = image['url']
-        max_resolution =  max(min_height, min_width)
+        num_of_underscores = url.count('_')
+        min_resolution = min(min_height, min_width)
         # Maping between the char that represents max resolution and the max resolution.
-        flickr_resolution_map = {2048: 'k', 1600: 'n', 1024: 'b', 800: 'c', 640: 'z',500: '',
-        400: 'w', 320: 'n', 240: 'm', 150: 'q', 100: 't', 75: 's'}
+        flickr_resolution_map = {
+                2048: 'k', 1600: 'n', 1024: 'b', 800: 'c', 640: 'z', 500: '',
+                400: 'w', 320: 'n', 240: 'm', 150: 'q', 100: 't', 75: 's'}
         for key in flickr_resolution_map:
-            if  max_resolution >= key:
-                if key == 500:
-                    # Removing the char that represents the resolution to get max resolution of 500.
-                    return  url[:-6] + url[-4:]
-                # Replacing the char that represents the resolution with the wonted key.
-                return  url[:-5] + flickr_resolution_map[key] + url[-4:]
+            # Url in the format: https://live.staticflickr.com/{server-id}/{id}_{secret}_{size-suffix}.jpg
+            if num_of_underscores == 2:
+                if min_resolution >= key:
+                    if key == 500:
+                        # Removing the char that represents the resolution to get max resolution of 500.
+                        return url[:-6] + url[-4:]
+                    # Replacing the char that represents the resolution with the wonted key.
```

Comment:
nit: wonted -> wanted

Reply:
fixed,
made small fixes to the function and added tests.